src/UseCases/HandlePayPalPaymentNotification/HandlePayPalPaymentNotificationUseCase.php

```diff
@@ -131,4 +143,51 @@ private function newPayPalDataFromRequest( PayPalNotificationRequest $request ):
 			->setPaymentTimestamp( $request->getPaymentTimestamp() );
 	}
 
+	private function donationWasBookedWithDifferentTransactionId( Donation $donation,
+																  PayPalNotificationRequest $request ): bool {
+		/**
+		 * @var PayPalPayment $payment
+		 */
+		$payment = $donation->getPaymentMethod();
+
+		if ( !$donation->isBooked() ) {
+			return false;
+		}
+
+		if ( $request->getTransactionId() === $payment->getPayPalData()->getPaymentId() ) {
+			return false;
+		}
+
+		if ( $payment->getPayPalData()->hasChildPayment( $request->getTransactionId() ) ) {
+			return false;
+		}
+
+		return true;
+	}
+
+	private function createChildDonation( Donation $donation, PayPalNotificationRequest $request ) {
+		$childPaymentMethod = new PayPalPayment( $this->newPayPalDataFromRequest( $request ) );
+		$payment = $donation->getPayment();
+		$childDonation = new Donation(
+			null,
+			Donation::STATUS_EXTERNAL_BOOKED,
+			$donation->getDonor(),
+			new DonationPayment( $payment->getAmount(), $payment->getIntervalInMonths(), $childPaymentMethod ),
+			$donation->getOptsIntoNewsletter(), $donation->getTrackingInfo()
+		);
+		try {
+			$this->repository->storeDonation( $childDonation );
+		} catch ( StoreDonationException $ex ) {
+			return;
+		}
+		/** @var PayPalPayment $paymentMethod */
+		$paymentMethod = $payment->getPaymentMethod();
+		$paymentMethod->getPayPalData()->addChildPayment( $request->getTransactionId(), $childDonation->getId() );
+		try {
+			$this->repository->storeDonation( $donation );
```

Comment:
So far, `DoctrineDonationRepository:storeDonation` does not support transactions with multiple donations. And in this case, that would not be useful IMHO, since we need to insert the child donation in the DB to generate its ID which then will be stored in the transaction data of the parent.